bentoml/adapters/multi_image_input.py

```diff
@@ -39,59 +50,96 @@ class MultiImageInput(BaseInputAdapter):
         ImportError: imageio package is required to use ImageInput
 
     Example usage:
-        ```
-        class MyService(BentoService):
-            @bentoml.api(input=MultiImageInput(input_names=('imageX', 'imageY')))
-            def predict(self, image_groups):
-                for image_group in image_groups:
-                    image_array_x = image_group['imageX']
-                    image_array_y = image_group['imageY']
-        ```
+
+    >>> class MyService(BentoService):
+    >>> @bentoml.api(input=MultiImageInput(input_names=('imageX', 'imageY')))
+    >>> def predict(self, image_groups):
+    >>>     for image_group in image_groups:
+    >>>         image_array_x = image_group['imageX']
+    >>>         image_array_y = image_group['imageY']
     """
 
     def __init__(
         self,
         input_names=("image",),
-        accept_image_formats=None,
+        accepted_image_formats=None,
         pilmode="RGB",
         is_batch_input=False,
         **base_kwargs,
     ):
         if is_batch_input:
-            raise ValueError('ImageInput can not accpept batch inputs')
+            raise ValueError('ImageInput can not accept batch inputs')
         super(MultiImageInput, self).__init__(**base_kwargs)
         self.input_names = input_names
         self.pilmode = pilmode
-        self.accept_image_formats = (
-            accept_image_formats or get_default_accept_image_formats()
+        self.accepted_image_formats = (
+            accepted_image_formats or get_default_accept_image_formats()
         )
 
-    def handle_batch_request(self, requests, func):
-        """Handles an HTTP request, convert it into corresponding data
-        format that user API function is expecting, and return API
-        function result as the HTTP response to client
-
-        :param requests: List of flask request object
-        :param func: user API function
-        """
-        raise NotImplementedError
+    def handle_request(self, request: Request, func):
+        files = {
+            name: self.read_file(file.filename, file.stream)
+            for (name, file) in request.files.items()
+        }
+        result = func((files,))[0]
+        return self.output_adapter.to_response(result, request)
+
+    def read_file(self, name: str, file):
+        safe_name = secure_filename(name)
+        verify_image_format_or_raise(safe_name, self.accepted_image_formats)
+        return imread(file, pilmode=self.pilmode)
+
+    def handle_batch_request(
+        self, requests: Iterable[SimpleRequest], func
+    ) -> Iterable[SimpleResponse]:
+        raise NotImplementedError(
```

Comment:
We should implement this handler first. The reason we made former ImageInput(LegacyImageInput) deprecated is we should implement this.

Reply:
https://github.com/bentoml/BentoML/blob/master/bentoml/adapters/image_input.py#L163

Take this as an example. When running with micro-batch enabled, batching server would queue some requests and pass them to `handle_batch_request` as a batch(list of SimpleRequest). The `handle_batch_request` is expected to handle these in efficient way, taking the advantage of batching.
https://github.com/tensorflow/serving/tree/master/tensorflow_serving/batching

Reply:
To decode multipart requests, we can use werkzeug Request like this: https://github.com/bentoml/BentoML/blob/master/bentoml/adapters/image_input.py#L175